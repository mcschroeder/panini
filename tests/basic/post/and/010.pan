and : (a:bool) -> (b:bool) -> {c:bool| c = true <=> (a = true /\ b = true)}

f10 : {x:bool|x=true} -> {y:bool|y=true} -> {z:bool|?}
f10 = \x:bool. \y:bool. and x y

f11 : {x:bool|x=true} -> {y:bool|y=false} -> {z:bool|?}
f11 = \x:bool. \y:bool. and x y

f12 : {x:bool|x=true} -> {y:bool|y=x} -> {z:bool|?}
f12 = \x:bool. \y:bool. and x y




f14 : {x:bool|x=true} -> {y:bool|true} -> {z:bool|?}
f14 = \x:bool. \y:bool. and x y
