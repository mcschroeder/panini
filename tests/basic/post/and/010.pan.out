f10 : {x:𝔹 | x = true} → {y:𝔹 | y = true} → {z:𝔹 | z = true}
f11 : {x:𝔹 | x = true} → {y:𝔹 | y = false} → {z:𝔹 | z = false}
f12 : {x:𝔹 | x = true} → {y:𝔹 | y = x} → {z:𝔹 | z = true}
f14 : {x:𝔹 | x = true} → {y:𝔹 | true} → {z:𝔹 | true}